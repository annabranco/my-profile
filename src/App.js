import React, { Component } from 'react';
import './styles/App.css';
import Header from './components/Header';
import MainArea from './components/MainArea';
import texts from './db/texts.js';

const doc = document.documentElement;
let adjustExpandedProjectsView = 0;

class App extends Component {
  	constructor(props) {
    	super(props)

    	this.state = {
	      	language: 'en',
	      	currentSection: {},
			doNotShowLanguagePopupAgain: undefined
    	}
  	}

  	componentDidMount() {
	    window.addEventListener('scroll', this.handleScroll);
	    this.loadDefaultLanguage();
  	}

	componentDidUpdate() {
		this.saveToLocalStorage();
	}

	componentWillUnmount() {
	 	window.removeEventListener('scroll', this.handleScroll);
	}


  	loadDefaultLanguage = () => {

	    if (localStorage.getItem("Anna Branco's professional profile") !== null) {
	      this.setState(
	        JSON.parse(localStorage.getItem("Anna Branco's professional profile"))
	      )
	    }
  	}

  	handleAdjustExpandedProjectsView = (adjust) => {
			adjustExpandedProjectsView = adjust;
	}

  	handleScroll = () => {

		if (doc.scrollTop >= 350 && doc.scrollTop <= 1540) {
			document.querySelector('.developer__formation').classList.add('comeIn');
		} else {
			document.querySelector('.developer__formation').classList.remove('comeIn');
		}

		if (doc.scrollTop >= 500 && doc.scrollTop <= 1540 + adjustExpandedProjectsView) {
			document.querySelector('.developer__projects').classList.add('comeIn');
		} else {
			document.querySelector('.developer__projects').classList.remove('comeIn');
		}

		if (doc.scrollTop >= 450 && doc.scrollTop <= 1540 + adjustExpandedProjectsView) {
			document.querySelector('.developer__sidebar').classList.add('comeIn');
		} else {
			document.querySelector('.developer__sidebar').classList.remove('comeIn');
		}

		if (doc.scrollTop >= 1050  + adjustExpandedProjectsView && doc.scrollTop <= 2000 + adjustExpandedProjectsView) {
			document.querySelector('.formation__horizontarBar-psy').classList.add('comeIn');
			document.querySelector('.formation__title-psy').classList.add('comeIn');
			for (const each of document.querySelectorAll('.formation__details-psy')){
				each.classList.add('comeIn');
			}
		} else {
			document.querySelector('.formation__horizontarBar-psy').classList.remove('comeIn');
			document.querySelector('.formation__title-psy').classList.remove('comeIn');
			for (const each of document.querySelectorAll('.formation__details-psy')){
				each.classList.remove('comeIn');
			}
		}

		if (doc.scrollTop >= 1200  + adjustExpandedProjectsView && doc.scrollTop <= 2000 + adjustExpandedProjectsView) {
			document.querySelector('.formation__horizontarBar-ir').classList.add('comeIn');
			document.querySelector('.formation__title-ir').classList.add('comeIn');
			for (const each of document.querySelectorAll('.formation__details-ir')){
				each.classList.add('comeIn');
			}
		} else {
			document.querySelector('.formation__horizontarBar-ir').classList.remove('comeIn');
			document.querySelector('.formation__title-ir').classList.remove('comeIn');
			for (const each of document.querySelectorAll('.formation__details-ir')){
				each.classList.remove('comeIn');
			}
		}

		if (doc.scrollTop >= 1320  + adjustExpandedProjectsView && doc.scrollTop <= 2000 + adjustExpandedProjectsView) {
			document.querySelector('.formation__horizontarBar-master').classList.add('comeIn');
			document.querySelector('.formation__title-master').classList.add('comeIn');
			for (const each of document.querySelectorAll('.formation__details-master')){
				each.classList.add('comeIn');
			}
		} else {
			document.querySelector('.formation__horizontarBar-master').classList.remove('comeIn');
			document.querySelector('.formation__title-master').classList.remove('comeIn');
			for (const each of document.querySelectorAll('.formation__details-master')){
				each.classList.remove('comeIn');
			}
		}

		if (doc.scrollTop >= 1420  + adjustExpandedProjectsView && doc.scrollTop <= 2000 + adjustExpandedProjectsView) {
			document.querySelector('.formation__horizontarBar-adalab').classList.add('comeIn');
			document.querySelector('.formation__title-adalab').classList.add('comeIn');


		} else {
			document.querySelector('.formation__horizontarBar-adalab').classList.remove('comeIn');
			document.querySelector('.formation__title-adalab').classList.remove('comeIn');

		}
  		console.log(doc.scrollTop);
  	}

  changeLanguage = event => {
    this.setState({
      language: event.currentTarget.lang
    });
  }

	clearLanguagePopup = doNotShowAgain => {
		this.setState({
			doNotShowLanguagePopupAgain: doNotShowAgain
		});
	}

	saveToLocalStorage = () => localStorage.setItem("Anna Branco's professional profile", JSON.stringify(this.state));

  render() {

    return (
      <div className="App">
	        <Header
				texts={texts}
				language={this.state.language}
				changeLanguage={this.changeLanguage}
			/>
	        <MainArea
		        handleAnimation={this.handleAnimation}
		        language={this.state.language}
		        changeLanguage={this.changeLanguage}
				texts={texts}
				clearLanguagePopup={this.clearLanguagePopup}
				doNotShowLanguagePopupAgain={this.state.doNotShowLanguagePopupAgain}
				handleAdjustExpandedProjectsView={this.handleAdjustExpandedProjectsView}
	        />
      </div>
    );
  }
}

export default App;
