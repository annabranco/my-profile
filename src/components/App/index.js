import React, { Component } from 'react';
import { Header, MainArea } from '../';
// import texts from '../../db/texts.js';

const doc = document.documentElement;
let adjustExpandedProjectsView = 0;

export class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      texts: this.props.texts,
      language: 'en',
      doNotShowLanguagePopupAgain: undefined,
      viewedAll: false
    };
  }

  componentDidMount() {
    window.addEventListener('scroll', this.handleScroll);
    // this.setTexts();
    this.loadDefaultLanguage();
    this.setState({ viewedAll: false });
    console.log('$$$ this.props.texts', this.props.texts);
  }

  componentDidUpdate() {
    this.saveToLocalStorage();
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.handleScroll);
  }

  // setTexts = () => {
  //   const { texts } = this.props;
  //   this.setState({ texts });
  // };

  loadDefaultLanguage = () => {
    if (localStorage.getItem("Anna Branco's professional profile") !== null) {
      this.setState(
        JSON.parse(localStorage.getItem("Anna Branco's professional profile"))
      );
    }
  };

  handleAdjustExpandedProjectsView = adjust => {
    adjustExpandedProjectsView = adjust;
  };

  handleScroll = () => {
    if (doc.scrollTop >= 350 && doc.scrollTop <= 1540) {
      document.querySelector('.developer__formation').classList.add('comeIn');
    } else {
      document
        .querySelector('.developer__formation')
        .classList.remove('comeIn');
    }

    if (
      doc.scrollTop >= 500 &&
      doc.scrollTop <= 1540 + adjustExpandedProjectsView
    ) {
      document.querySelector('.developer__projects').classList.add('comeIn');
    } else {
      document.querySelector('.developer__projects').classList.remove('comeIn');
    }

    if (
      doc.scrollTop >= 450 &&
      doc.scrollTop <= 1540 + adjustExpandedProjectsView
    ) {
      document.querySelector('.developer__sidebar').classList.add('comeIn');
    } else {
      document.querySelector('.developer__sidebar').classList.remove('comeIn');
    }

    if (
      doc.scrollTop >= 1050 + adjustExpandedProjectsView &&
      doc.scrollTop <= 2000 + adjustExpandedProjectsView
    ) {
      document
        .querySelector('.formation__horizontarBar-psy')
        .classList.add('comeIn');
      document.querySelector('.formation__title-psy').classList.add('comeIn');
      for (const each of document.querySelectorAll('.formation__details-psy')) {
        each.classList.add('comeIn');
      }
    } else {
      document
        .querySelector('.formation__horizontarBar-psy')
        .classList.remove('comeIn');
      document
        .querySelector('.formation__title-psy')
        .classList.remove('comeIn');
      for (const each of document.querySelectorAll('.formation__details-psy')) {
        each.classList.remove('comeIn');
      }
    }

    if (
      doc.scrollTop >= 1200 + adjustExpandedProjectsView &&
      doc.scrollTop <= 2000 + adjustExpandedProjectsView
    ) {
      document
        .querySelector('.formation__horizontarBar-ir')
        .classList.add('comeIn');
      document.querySelector('.formation__title-ir').classList.add('comeIn');
      for (const each of document.querySelectorAll('.formation__details-ir')) {
        each.classList.add('comeIn');
      }
    } else {
      document
        .querySelector('.formation__horizontarBar-ir')
        .classList.remove('comeIn');
      document.querySelector('.formation__title-ir').classList.remove('comeIn');
      for (const each of document.querySelectorAll('.formation__details-ir')) {
        each.classList.remove('comeIn');
      }
    }

    if (
      doc.scrollTop >= 1320 + adjustExpandedProjectsView &&
      doc.scrollTop <= 2000 + adjustExpandedProjectsView
    ) {
      document
        .querySelector('.formation__horizontarBar-master')
        .classList.add('comeIn');
      document
        .querySelector('.formation__title-master')
        .classList.add('comeIn');
      for (const each of document.querySelectorAll(
        '.formation__details-master'
      )) {
        each.classList.add('comeIn');
      }
    } else {
      document
        .querySelector('.formation__horizontarBar-master')
        .classList.remove('comeIn');
      document
        .querySelector('.formation__title-master')
        .classList.remove('comeIn');
      for (const each of document.querySelectorAll(
        '.formation__details-master'
      )) {
        each.classList.remove('comeIn');
      }
    }

    if (
      doc.scrollTop >= 1420 + adjustExpandedProjectsView &&
      doc.scrollTop <= 2000 + adjustExpandedProjectsView
    ) {
      document
        .querySelector('.formation__horizontarBar-adalab')
        .classList.add('comeIn');
      document
        .querySelector('.formation__title-adalab')
        .classList.add('comeIn');
    } else {
      document
        .querySelector('.formation__horizontarBar-adalab')
        .classList.remove('comeIn');
      document
        .querySelector('.formation__title-adalab')
        .classList.remove('comeIn');
    }
    console.log(doc.scrollTop);
  };

  changeLanguage = event =>
    this.setState({ language: event.currentTarget.lang });

  clearLanguagePopup = doNotShowAgain =>
    this.setState({ doNotShowLanguagePopupAgain: doNotShowAgain });

  saveToLocalStorage = () =>
    localStorage.setItem(
      "Anna Branco's professional profile",
      JSON.stringify(this.state)
    );

  userViewedAllComponents = () => this.setState({ viewedAll: true });

  render() {
    const { language } = this.state;
    const { texts } = this.props;

    return (
      <div className="App">
        <Header
          texts={texts[language].header}
          changeLanguage={this.changeLanguage}
        />
        <MainArea
          changeLanguage={this.changeLanguage}
          texts={texts[language]}
          clearLanguagePopup={this.clearLanguagePopup}
          doNotShowLanguagePopupAgain={this.state.doNotShowLanguagePopupAgain}
          handleAdjustExpandedProjectsView={
            this.handleAdjustExpandedProjectsView
          }
          userViewedAllComponents={this.userViewedAllComponents}
          viewedAll={this.state.viewedAll}
        />
      </div>
    );
  }
}
