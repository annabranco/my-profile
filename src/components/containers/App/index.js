import React, { Component, Fragment } from 'react';
import { MainArea } from '../';
import { Header } from '../../views';
import { getLanguageCodeByName } from '../../../utils/languages';

const DEFAULT_PAGE_LANGUAGE = 'English';

export class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      texts: props.texts,
      language: getLanguageCodeByName(DEFAULT_PAGE_LANGUAGE),
      doNotShowLanguageModalAgain: undefined,
      displayThanksMessage: false
    };
  }

  componentDidMount() {
    this.loadLanguageSettings();
  }

  saveLanguageSettings = () => {
    const { language, doNotShowLanguageModalAgain } = this.state;
    localStorage.setItem(
      "Anna Branco's professional profile",
      JSON.stringify({
        language,
        doNotShowLanguageModalAgain
      })
    );
  };

  loadLanguageSettings = () => {
    if (localStorage.getItem("Anna Branco's professional profile") !== null) {
      const { language, doNotShowLanguageModalAgain } = JSON.parse(
        localStorage.getItem("Anna Branco's professional profile")
      );
      this.setState({ language, doNotShowLanguageModalAgain });
    }
  };

  onChangeLanguage = event =>
    this.setState({ language: event.currentTarget.lang });

  closeLanguageModal = willModalBeDisplayedAgain =>
    this.setState(
      { doNotShowLanguageModalAgain: willModalBeDisplayedAgain },
      () => this.saveLanguageSettings()
    );

  triggerThankYouMessage = () => this.setState({ displayThanksMessage: true });

  render() {
    const { language } = this.state;
    const { texts, APP_VERSION } = this.props;

    return (
      <Fragment>
        <Header
          texts={texts[language].header}
          language={language}
          onChangeLanguage={this.onChangeLanguage}
          APP_VERSION={APP_VERSION}
        />
        <MainArea
          onChangeLanguage={this.onChangeLanguage}
          texts={texts[language]}
          closeLanguageModal={this.closeLanguageModal}
          doNotShowLanguageModalAgain={this.state.doNotShowLanguageModalAgain}
          handleAdjustExpandedProjectsView={
            this.handleAdjustExpandedProjectsView
          }
          triggerThankYouMessage={this.triggerThankYouMessage}
          displayThanksMessage={this.state.displayThanksMessage}
        />
      </Fragment>
    );
  }
}
