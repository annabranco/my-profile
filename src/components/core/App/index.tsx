import { useEffect, ReactElement } from 'react';
import { ILanguagePreferences } from '../../../types/interfaces';
import ErrorBoundary from '../ErrorBoundary';
import { useStateWithLabel } from '../../../utils/hooks';
import Header from '../../mainComponents/Header';
import LanguagesModal from '../../elements/LanguagesModal';
import MainArea from '../../mainComponents/MainArea';
import {
  HIDE_FOREVER,
  LANG_MODAL_VISIBLE,
  SETTINGS_LOADED
} from '../../../constants';
import { GlobalStyles } from '../../../styles/global';

const App = (): ReactElement => {
  const languageSettings: ILanguagePreferences | Record<string, never> =
    JSON.parse(
      localStorage.getItem("Anna Branco's professional profile") as string
    ) || {};

  const [hideForever, toggleHideForever] = useStateWithLabel<boolean>(
    languageSettings.hideLanguagesModalForever || false,
    HIDE_FOREVER
  );
  const [settingsLoaded, setSettingsLoaded] = useStateWithLabel<boolean>(
    false,
    SETTINGS_LOADED
  );
  const [langModalVisible, toggleModalVisible] = useStateWithLabel<boolean>(
    false,
    LANG_MODAL_VISIBLE
  );
  const [isSeabedElementOpened, openSeabedElement] = useStateWithLabel<boolean>(
    false,
    'isSeabedElementOpened'
  );
  const toggleBlockLangModal = (): void =>
    toggleHideForever(prevState => !prevState);

  const onCloseLanguageModal = (): void => {
    toggleModalVisible(false);
    toggleHideForever(hideForever);
    // setTimeout(() => {
    // // TODO: Review modal button clicking on e2e test. It is not closing modal.
    //   console.log('$$$ langModalVisible', langModalVisible);
    // }, 3000);
  };

  useEffect(() => {
    const loadLanguageSettings = (): void => {
      toggleHideForever(
        languageSettings.hideLanguagesModalForever || hideForever
      );
      setSettingsLoaded(true);
    };
    loadLanguageSettings();
    if (!hideForever) {
      toggleModalVisible(true);
    }
  }, [
    hideForever,
    languageSettings.hideLanguagesModalForever,
    setSettingsLoaded,
    toggleHideForever,
    toggleModalVisible
  ]);

  return (
    <ErrorBoundary>
      <GlobalStyles />
      {langModalVisible && (
        <LanguagesModal
          hideForever={hideForever}
          onCloseLanguageModal={onCloseLanguageModal}
          toggleBlockLangModal={toggleBlockLangModal}
        />
      )}
      <Header
        hideForever={hideForever}
        isSeabedElementOpened={isSeabedElementOpened}
      />
      {settingsLoaded && (
        <MainArea
          isSeabedElementOpened={isSeabedElementOpened}
          langModalVisible={langModalVisible}
          openSeabedElement={openSeabedElement}
        />
      )}
    </ErrorBoundary>
  );
};

export default App;
