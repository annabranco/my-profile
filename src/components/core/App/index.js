import React, { useEffect } from 'react';
import { arrayOf, string } from 'prop-types';
import { useStateWithLabel } from '../../../utils/hooks';
import { getLanguageCodeByName } from '../../../utils/languages';
import ErrorBoundary from '../ErrorBoundary';
import { MainArea } from '../../containers';
import { Header } from '../../views';
import {
  BLOCK_LANG_MODAL,
  DISPLAY_THANKS,
  LANGUAGE,
  SETTINGS_LOADED
} from '../../../constants';
import {
  allLanguagesTextsPropType,
  experiencesTextPropType,
  formationPropType,
  languagesPropType,
  projectsPropType,
  skillGroupsPropType
} from '../../../types';

const DEFAULT_PAGE_LANGUAGE = 'English';

const App = ({
  APP_VERSION,
  experiences,
  formation,
  languages,
  projects,
  skills,
  texts
}) => {
  const [blockLangModal, toggleBlockModal] = useStateWithLabel(
    false,
    BLOCK_LANG_MODAL
  );
  const [displayThanks, setDisplayThanks] = useStateWithLabel(
    false,
    DISPLAY_THANKS
  );
  const [language, setLanguage] = useStateWithLabel(
    getLanguageCodeByName(DEFAULT_PAGE_LANGUAGE),
    LANGUAGE
  );
  const [settingsLoaded, setSettingsLoaded] = useStateWithLabel(
    false,
    SETTINGS_LOADED
  );

  const onChangeLanguage = event => setLanguage(event.currentTarget.lang);

  const closeLanguageModal = hideForever => {
    toggleBlockModal(hideForever);
    localStorage.setItem(
      "Anna Branco's professional profile",
      JSON.stringify({
        language,
        hideLanguagesModalForever: hideForever
      })
    );
  };

  const triggerThankYouMessage = () => setDisplayThanks(true);

  useEffect(() => {
    const loadLanguageSettings = () => {
      const languageSettings =
        JSON.parse(
          localStorage.getItem("Anna Branco's professional profile")
        ) || {};
      toggleBlockModal(
        languageSettings.hideLanguagesModalForever || blockLangModal
      );
      setSettingsLoaded(true);
    };

    loadLanguageSettings();
  }, [blockLangModal, language, setSettingsLoaded, toggleBlockModal]);

  return (
    <ErrorBoundary texts={texts[language].error}>
      <Header
        APP_VERSION={APP_VERSION}
        language={language}
        languages={languages}
        onChangeLanguage={onChangeLanguage}
        texts={texts[language].header}
      />
      {settingsLoaded && (
        <MainArea
          blockLangModal={blockLangModal}
          closeLanguageModal={closeLanguageModal}
          displayThanks={displayThanks}
          experiences={experiences}
          formation={formation}
          language={language}
          languages={languages}
          onChangeLanguage={onChangeLanguage}
          projects={projects}
          skills={skills}
          texts={texts[language]}
          triggerThankYouMessage={triggerThankYouMessage}
        />
      )}
    </ErrorBoundary>
  );
};

App.propTypes = {
  APP_VERSION: string.isRequired,
  experiences: arrayOf(experiencesTextPropType).isRequired,
  formation: arrayOf(formationPropType).isRequired,
  languages: arrayOf(languagesPropType).isRequired,
  projects: arrayOf(projectsPropType).isRequired,
  skills: arrayOf(skillGroupsPropType).isRequired,
  texts: allLanguagesTextsPropType.isRequired
};

export default App;
