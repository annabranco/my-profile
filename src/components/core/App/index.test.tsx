import { shallow, ShallowWrapper } from 'enzyme';
import 'jest-styled-components';
import '../../../tests/mocks/matchMedia';
import App from '.';
import { setupMockProvider } from '../../../tests/mocks';

// TODO [10-May -21]: Implement defined testing suite (Anna Branco)

describe('< App >', () => {
  let MockProvider;
  let wrapper: ShallowWrapper;

  beforeAll(() => {
    MockProvider = setupMockProvider({ language: 'en' }).MockProvider;
    wrapper = shallow(
      <MockProvider>
        <App />
      </MockProvider>
    );
  });

  it('should mount', () => {
    expect(wrapper).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  xit('should have an Error Boundary', () => {});

  xit('should get languages preferences from localStorage', () => {});

  xit('should display a language modal on mount', () => {});

  xit('should render a Header component', () => {});

  xit('should display a Main Area component only when settings are loaded', () => {});
});
