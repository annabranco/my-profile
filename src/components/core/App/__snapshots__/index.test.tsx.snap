// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< App > matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <App />
</Provider>
`;
